### PR TITLE
Docs: Handle Throwing Expression in SpeziLLMOpenAI.md's LLMOpenAIDemo Example

### PR DESCRIPTION
**Pull request by paulhdk**
# Handle Throwing Expression in SpeziLLMOpenAI.md's LLMOpenAIDemo Example

## :recycle: Current situation & Problem

After copying a code example from the SpeziLLM documentation, I noticed that ...
-  a throwing expression wasn’t being handled. To make life eseier for future readers, add a simple `do … catch` for handling any exceptions that may be thrown. Also, happy to take suggestions for a more elegant way to do this!
- some imports were missing. To make life eseier for future readers, add the missing imports. I hope I didn’t miss any.
- I was getting an error around the use of `@EnvironmentObject`. Fix the error by substituting `@EnvironmentObject` with `@Environment`, which is also in line with the code in `TestApp`.

## :gear: Release Notes 
- Update documentation

## :books: Documentation
- Update Documentation

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,8 +127,12 @@ struct LLMLocalDemoView: View {
...
                 )
 
-                for try await token in try await llmSession.generate() {
-                    responseText.append(token)
+                do {
+                    for try await token in try await llmSession.generate() {
+                        responseText.append(token)
+                    }
+                } catch {
+                    // Handle errors ...
```
Comment (PSchmiedmayer) [inline README.md:135]: We could add a small tip here to look into `ViewState` as a nice way to handle errors in a simple setup: https://swiftpackageindex.com/stanfordspezi/speziviews/1.6.0/documentation/speziviews#manage-and-communicate-view-state. We should add a documentation to the main [`ViewState`](https://swiftpackageindex.com/stanfordspezi/speziviews/1.6.0/documentation/speziviews/viewstate) type, we only have a larger example for [`OperationState`](https://swiftpackageindex.com/stanfordspezi/speziviews/1.6.0/documentation/speziviews/operationstate)

We could use a [`> [!TIP]`](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) in the MD files and the similar setup in the [DocC syntax](https://www.swift.org/documentation/docc/other-formatting-options#Add-Notes-and-Other-Asides).

File: Sources/SpeziLLMOpenAI/SpeziLLMOpenAI.docc/SpeziLLMOpenAI.md [hunk 2]
```diff
@@ -125,10 +137,12 @@ The ``LLMOpenAIAPITokenOnboardingStep`` provides a view that can be used for the
 First, create a new view to show the onboarding step:
 
 ```swift
+import SpeziLLMOpenAI
 import SpeziOnboarding
+import SwiftUI
 
 struct OpenAIAPIKey: View {
-    @EnvironmentObject private var onboardingNavigationPath: OnboardingNavigationPath
+    @Environment(OnboardingNavigationPath.self) private var onboardingNavigationPath: OnboardingNavigationPath
```
Comment (PSchmiedmayer) [inline Sources/SpeziLLMOpenAI/SpeziLLMOpenAI.docc/SpeziLLMOpenAI.md:145]: Thank you for fixing this in the documentation here!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @paulhdk! I have merged the different documentation branches into a single one to group them & give a single review. It think it's fine to merge them in a single PR as they all aim for the same goal 👍

I only had one hint that we could point people to `ViewState` to simply error handling. We don't need to add a tip for every element but we could add it in the inside code comment, e.g. like:
```swift
// Handle errors here. E.g., you can use `ViewState` and `viewStateAlert` from Spezi Views.
```

File: README.md (hunk 1 above)
Comment (paulhdk) [reply README.md]: @PSchmiedmayer PTAL. I've updated the comments here, as requested above, and have pushed some `ViewState` documentation to https://github.com/StanfordSpezi/SpeziViews/pull/43 for us to iterate on.

Comment (PSchmiedmayer) [reply README.md]: Thank you; looks great!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements; the changes look great!

**Pull request merged**